lewis_emulators/fins/interfaces/stream_interface.py

```diff
@@ -72,24 +72,21 @@ def any_command(self, command):
         # The helium recovery PLC memory map has addresses that store types that take up either one word (16 bits) or
         # two. Most take up one word, so if the number of words to read is two we check that the client wants to read
         # from a memory location from where a 32 bit value starts.
-        if number_of_words_to_read == 2 and memory_start_address not in self.device.int32_memory.keys():
+        if number_of_words_to_read == 2 and (memory_start_address not in self.device.int32_memory.keys() and
+                                             memory_start_address not in self.device.float_memory.keys()):
             raise ValueError("The memory start address {} corresponds to a single word in the memory map, "
                              "not two.".format(memory_start_address))
         # The PLC also stores 32 bit floating point numbers, but the asyn device support makes the IOC ask for 4 bytes
         # instead of two.
-        elif number_of_words_to_read == 4 and memory_start_address not in self.device.float_memory.keys():
-            raise ValueError("The only commands that should ask for 4 words are for reading memory locations that "
-                             "store real numbers, which {} is not.".format(memory_start_address))
-        elif number_of_words_to_read > 4 or number_of_words_to_read == 3:
-            raise ValueError("The memory map only specifies data types for which commands should ask for one, two or "
-                             "four words at most .")
-
+        elif number_of_words_to_read > 2:
+            raise ValueError("The memory map only specifies data types for which commands should ask for one or two at most.")
+        is_float = True if memory_start_address in self.device.float_memory.keys() else False
```

Comment:
Suggestion for future, but not important for this ticket

```suggestion
        is_float = memory_start_address in self.device.float_memory.keys()
```

Reply:
very "True" it is already a boolean :-D